systems/common/msgBusServiceClient/msgbus_service_client.go

```diff
@@ -0,0 +1,151 @@
+package msgBusServiceClient
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/sirupsen/logrus"
+
+	pb "github.com/ukama/ukama/systems/init/msgClient/pb/gen"
+	"google.golang.org/grpc"
+)
+
+type MsgBusServiceClient interface {
+	Register() error
+	Start() error
+	Stop() error
+	PublishRequest(route string, msg protoreflect.ProtoMessage) error
+}
+
+type msgBusServiceClient struct {
+	uuid         string
+	service      string
+	system       string
+	instanceId   string
+	msgBusURI    string
+	msgClientURI string
+	exchange     string
+	listQueue    string
+	publQueue    string
+	conn         *grpc.ClientConn
+	client       pb.MsgClientServiceClient
+	timeout      time.Duration
+	host         string
+	retry        int8
+	routes       []string
+}
+
+func NewMsgBusClient(timeout time.Duration, system string,
+	service string, instanceId string, msgBusURI string,
+	serviceURI string, msgClientURI string, exchange string, lq string, pq string, retry int8, routes []string) *msgBusServiceClient {
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, msgClientURI, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		logrus.Fatalf("did not connect: %v", err)
+	}
+	client := pb.NewMsgClientServiceClient(conn)
+
+	return &msgBusServiceClient{
+		service:      service,
+		system:       system,
+		instanceId:   instanceId,
+		msgBusURI:    msgBusURI,
+		msgClientURI: msgClientURI,
+		conn:         conn,
+		client:       client,
+		timeout:      timeout,
+		retry:        retry,
+		host:         serviceURI,
+		routes:       routes,
+		listQueue:    lq,
+		publQueue:    pq,
+		exchange:     exchange,
+	}
+
+}
+
+func (m *msgBusServiceClient) Register() error {
+	logrus.Debugf("Registering %s service instance %s to MessageBusClient at %s.", m.service, m.instanceId, m.msgClientURI)
+	ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+	defer cancel()
+
+	resp, err := m.client.RegisterService(ctx, &pb.RegisterServiceReq{
+		ServiceName: m.service,
```

Comment:
I think you need to verify that all fields of m *msgBusServiceClient have values before use them in the req .and check the REGISTRATION_STATUS  to handle different states.what happened when a service is already registered ?I think we should show some type of logs and not ovewrite the old instance.

Reply:
Service will register once in it's life time. I don see any use case where service woul re-register itself. For upgardes or crash  , service would anyway request for registration again on startup. In that all new parameters will be taken by msgClient sdervice but the UUID provided with the previous existing one.  let me know if I missed anything or  you have any use case where we would need it.

Reply:
Also check is avaibale for registration afetr error handling